### PR TITLE
FEAT(#009): DESIGN PAGE FOR MATERIALS MANAGEMENT

### LINKED ISSUE #9: Diseño de página para la gestión de materiales de un usuario

### PR DESCRIPTION
**Pull request by mrarjona**
I have completed the graphical implementation of the user materials management page. Now, the page displays in a clear and organized way all the materials associated to a user, with intuitive options to download, edit and delete each one. I have incorporated creation and editing forms for a smoother experience.

Additionally, I have integrated a pagination that displays 5 items per page, improving navigation in cases of large sets of materials. I have also added the ability to filter the list by title and price, allowing users to quickly find the materials they are looking for.

**For the reviewer**

Check if it complies with the established:

- [x] Confirm that all materials associated with a user are correctly listed on the page.
- [x] Test the correct display of the creation and editing forms for adding and modifying materials.
- [x] Confirm that pagination is implemented, showing 5 materials per page.
- [x] Test the filtering mechanism for materials based on title and price.
- [x] Verify that the filter results are accurate and update dynamically.
- [x] Check the responsiveness of the page on various devices and screen sizes.
- [x] Ensure proper error messages are displayed for invalid inputs or actions.
- [x] Review the code for readability, maintainability, and adherence to coding standards.

### CODE REVIEW

File: client/src/pages/materials.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from 'react'
```
Comment (ivamorgra) [inline client/src/pages/materials.tsx]: Delete this unuse import please

File: client/src/components/materials-files/materials-files.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from 'react'
```
Comment (ivamorgra) [inline client/src/components/materials-files/materials-files.tsx]: delete it please

File: client/src/components/materials-filter/materials-filter.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react'
```
Comment (ivamorgra) [inline client/src/components/materials-filter/materials-filter.tsx]: delete it xd

File: client/src/components/pagination/pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react'
```
Comment (ivamorgra) [inline client/src/components/pagination/pagination.tsx]: manin, same

**Review by ivamorgra — DISMISSED** [review]
Please, delete these unusual variables before merge to develop. Good Job!

**Comment by mrarjona** [comment]
> Please, delete these unusual variables before merge to develop. Good Job!

Perfect @ivamorgra, thank you very much! I have implemented all the suggestions. 🤝

**Review by TomasCB24 — APPROVED** [review]
All ok Miguel.

**Pull request merged**